packages/features/auth/lib/next-auth-options.ts

```diff
@@ -391,16 +391,35 @@ export const AUTH_OPTIONS: AuthOptions = {
             name: true,
             email: true,
             role: true,
+            teams: {
+              include: {
+                team: true,
+              },
+            },
           },
         });
 
         if (!existingUser) {
           return token;
         }
 
+        // Check if the existingUser has any active teams
+        const belongsToActiveTeam =
+          existingUser.teams.filter((m: { team: { metadata: unknown } }) => {
+            if (!IS_TEAM_BILLING_ENABLED) return true;
+            const metadata = teamMetadataSchema.safeParse(m.team.metadata);
+            if (metadata.success && metadata.data?.subscriptionId) return true;
+            return false;
+          }).length > 0;
+
         return {
-          ...existingUser,
+          id: existingUser.id,
```

Comment:
cc @emrysal @zomars This is a critical place to change, looks good to me, but you should check too.

Reply:
I'm not sure why we shouldn't continue to just ...existingUser, belongsToActiveTeam will append. 💭